### PR TITLE
core: services: cable_guy: Move interface priority from ipr to dhcpcd

### LINKED ISSUE #1539: Allow changing network priority
When using a vehicle with multiple network interfaces: 4G, satellite, ethernet, wifi. We should allow the user to choose the priority between each one.

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #1539

Right now to have it working it's required to do a system reboot.
Doing dhcpcd service restart breaks our wpa_supplicant connection. #1958 was an attempt to fix that.

### CODE REVIEW

File: core/frontend/src/components/app/NetworkInterfacePriorityMenu.vue [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <v-list>
       <v-card-subtitle class="text-md-center" max-width="30">
         Move network interfaces over
-        to change network access priority
+        to change network access priority.<br>Applied changes require a <b>system reboot</b>.
```
Comment (Williangalvani) [inline core/frontend/src/components/app/NetworkInterfacePriorityMenu.vue:6]: can we make it show the reboo-required icon we use for parameter changes? even in bold this is quite subtle.
a popup after saving saying "configurations will be applied on the next boot" should work well, too.

File: core/services/cable_guy/api/manager.py [hunk 2]
```diff
@@ -396,71 +415,182 @@ def get_interfaces_priority(self) -> List[NetworkInterfaceMetric]:
...
+
+        List[NetworkInterfaceMetric]: A list of priority metrics for each interface.
         """
-        metric: NetworkInterfaceMetric
-        for metric in self.get_interfaces_priority():
-            if interface_name == metric.name:
-                return metric
+        lines = self._get_service_dhcpcd_content()
+        result = []
+        current_interface = None
```
Comment (Williangalvani) [inline core/services/cable_guy/api/manager.py:476]: did you miss the type here? it looks like you missed the type, and that caused the errors in line 480 and 482 that you suppressed.

File: core/services/cable_guy/api/manager.py [hunk 3]
```diff
@@ -396,71 +415,182 @@ def get_interfaces_priority(self) -> List[NetworkInterfaceMetric]:
...
+                sorted by priority to set, if values are undefined.
+        """
+        self._set_interfaces_priority_to_dhcpcd(interfaces)
+
+    def _set_interfaces_priority_to_dhcpcd(self, interfaces: List[NetworkInterfaceMetricApi]) -> None:
+        """Sets network interface priority..
 
         Args:
-            interfaces (List[NetworkInterfaceMetricApi]): A list of interfaces and their priority metrics.
+            interfaces (List[NetworkInterfaceMetricApi]): List of network interfaces
```
Comment (Williangalvani) [inline core/services/cable_guy/api/manager.py:547]: the priority metrics are still there, why was the description changed?

File: core/services/cable_guy/api/manager.py [hunk 4]
```diff
@@ -396,71 +415,182 @@ def get_interfaces_priority(self) -> List[NetworkInterfaceMetric]:
             else:
                 metric_dict[d["index"]] = d["metric"]
 
+        # Highest priority wins for ipr but not for dhcpcd, so we sort and revet the list
```
Comment (Williangalvani) [inline core/services/cable_guy/api/manager.py:418]:
```suggestion
        # Highest priority wins for ipr but not for dhcpcd, so we sort and reverse the list
```

File: core/frontend/src/components/app/NetworkInterfacePriorityMenu.vue (hunk 1 above)
Comment (patrickelectric) [reply core/frontend/src/components/app/NetworkInterfacePriorityMenu.vue:6]: > can we make it show the reboo-required icon we use for parameter changes? even in bold this is quite subtle. Or maybe a popup after saving saying "configurations will be applied on the next boot", if that is easier.

Yes, but that idea is under https://github.com/bluerobotics/BlueOS/pull/1962, to be a global notifier.

**Review by Williangalvani — CHANGES_REQUESTED** [review]
It is working well, nice. Just had some minor questions

**Comment by Williangalvani** [comment]
btw, I tested with `ip route get <external ip>`
that shows what interface the packet would be routed through to get at the destination

File: core/services/cable_guy/api/manager.py (hunk 2 above)
Comment (patrickelectric) [reply core/services/cable_guy/api/manager.py:476]: Yes, this is unrelated, the source of the issue is from mypy.
https://github.com/python/mypy/issues/14987

**Comment by joaoantoniocardoso** [comment]
Here it's not changing the metrics, and `ip route get` is always trying through the same interface (eth0)

**Comment by patrickelectric** [comment]
> Here it's not changing the metrics (seeing using `ip -4 route` and `ip -6 route`), and `ip route get` is always trying through the same interface (eth0)

did you reboot ?

**Comment by joaoantoniocardoso** [comment]
> > Here it's not changing the metrics (seeing using `ip -4 route` and `ip -6 route`), and `ip route get` is always trying through the same interface (eth0)
> 
> did you reboot ?

I did not. Oops

**Comment by joaoantoniocardoso** [comment]
https://github.com/bluerobotics/BlueOS/assets/5920286/c33f34d1-d39f-4d3b-bc0b-eee8dc7f3ff1

**Review by joaoantoniocardoso — APPROVED** [review]

**Pull request merged**